python-avd/pyavd/_eos_designs/structured_config/network_services/port_channel_interfaces.py

```diff
@@ -20,122 +23,93 @@ class PortChannelInterfacesMixin(Protocol):
-                        if (short_esi := endpoint.port_channel.short_esi) is not None and len(short_esi.split(":")) == 3:
-                            interface.update(
-                                {
-                                    "evpn_ethernet_segment": {
-                                        "identifier": f"{self.inputs.evpn_short_esi_prefix}{short_esi}",
-                                        "route_target": short_esi_to_route_target(short_esi),
-                                    },
-                                },
-                            )
-                            if port_channel_mode == "active":
-                                interface["lacp_id"] = short_esi.replace(":", ".")
-
-                        append_if_not_duplicate(
-                            list_of_dicts=port_channel_interfaces,
-                            primary_key="name",
-                            new_dict=interface,
-                            context="Port-Channel Interfaces defined under point_to_point_services",
-                            context_keys=["name"],
+            self._set_point_to_point_interfaces(tenant, subif_parent_interfaces)
+
+            for subif_parent_interface in subif_parent_interfaces:
+                if subif_parent_interface.name not in self.structured_config.port_channel_interfaces:
+                    self.structured_config.port_channel_interfaces.append(subif_parent_interface)
+
+    def _set_point_to_point_interfaces(
+        self: AvdStructuredConfigNetworkServicesProtocol,
+        tenant: EosDesigns._DynamicKeys.DynamicNetworkServicesItem.NetworkServicesItem,
+        subif_parent_interfaces: list[EosCliConfigGen.PortChannelInterfacesItem],
+    ) -> None:
+        """Set the structured_config port_channel_interfaces with the point-to-point interfaces defined under network_services."""
+        for point_to_point_service in tenant.point_to_point_services._natural_sorted():
+            for endpoint in point_to_point_service.endpoints:
+                if self.shared_utils.hostname not in endpoint.nodes:
+                    continue
+
+                node_index = endpoint.nodes.index(self.shared_utils.hostname)
+                interface_name = endpoint.interfaces[node_index]
+                if (port_channel_mode := endpoint.port_channel.mode) not in ["active", "on"]:
+                    continue
+
+                channel_group_id = "".join(re.findall(r"\d", interface_name))
+                interface_name = f"Port-Channel{channel_group_id}"
+                if point_to_point_service.subinterfaces:
+                    # This is a subinterface so we need to ensure that the parent is created
+                    parent_interface = EosCliConfigGen.PortChannelInterfacesItem(
+                        name=interface_name,
+                        peer_type="system",
+                        shutdown=False,
+                    )
+                    parent_interface.switchport.enabled = False
```

Comment:
Let's finish the ethernet PR and then implement the parent interface logic the same here.

Reply:
Addressed differently as discussed